Source/WebCore/platform/network/curl/CurlMultipartHandle.h

```diff
@@ -39,39 +40,51 @@ class SharedBuffer;
 class CurlMultipartHandle {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    static std::unique_ptr<CurlMultipartHandle> createIfNeeded(CurlMultipartHandleClient&, const CurlResponse&);
+    WEBCORE_EXPORT static std::unique_ptr<CurlMultipartHandle> createIfNeeded(CurlMultipartHandleClient&, const CurlResponse&);
 
-    CurlMultipartHandle(CurlMultipartHandleClient&, const String&);
+    CurlMultipartHandle(CurlMultipartHandleClient&, CString&&);
     ~CurlMultipartHandle() { }
 
-    void didReceiveData(const SharedBuffer&);
-    void didComplete();
+    WEBCORE_EXPORT void didReceiveMessage(const SharedBuffer&);
```

Comment:
Looking through the test code, there are a lot of `SharedBuffer::create` unnecessarily. I think didReceiveMessage should take `std::span<uint8_t>`. WDYT?

Reply:
I want to change it to `const uint8_t* data, size_t size`.

As a premise, I want to stop creating `SharedBuffer` in `CurlRequest::didReceiveDataCallback`. This is because if the `CURL_WRITEFUNC_PAUSE` condition is entered in `CurlRequest::didReceiveData`, this SharedBuffer will not be used and will only be discarded. In other words, there are cases where SharedBuffer is generated unnecessarily.

Instead, I would like to change the function argument as shown below to pass the buffer pointer notified by libcurl. By doing so, you can reduce unnecessary data type conversions.

```
 CurlRequest::didReceiveData(const uint8_t* data, size_t size);
 CurlMultipartHandle::didReceiveData(const uint8_t* data, size_t size);
```

I will create a PR to make this change in the future.